builder/dockerfile/dispatchers.go

```diff
@@ -414,6 +414,8 @@ func cmd(b *Builder, args []string, attributes map[string]bool, original string)
 	}
 
 	b.runConfig.Cmd = strslice.StrSlice(cmdSlice)
+	// set config as already being escaped, this prevents double escaping on windows
+	b.runConfig.ArgsEscaped = true
```

Comment:
Ah. Perhaps RUN is broken too, then.

What happens if you write:

`RUN ["cmd /c echo hi"]`

What should happen is that it should fail with file not found, but I bet it succeeds.